flink-connectors/flink-connector-cassandra/pom.xml

```diff
@@ -36,9 +36,11 @@ under the License.
 
 	<!-- Allow users to pass custom connector versions -->
 	<properties>
-		<cassandra.version>2.2.5</cassandra.version>
-		<driver.version>3.0.0</driver.version>
-		<guava.version>18.0</guava.version>
+		<!-- Cassandra version 4.x allow to address clusters of version 3.x as well -->
+		<cassandra.version>4.0.3</cassandra.version>
+		<!--driver 3.x works with 3.x and 4.x versions of Cassandra but driver 4.x is a complete refactoring with different API-->
+		<driver.version>3.11.2</driver.version>
```

Comment:
You should be able to safely exclude the jnr-ffi and jnr-posix dependencies.  The driver uses jnr-posix to access native calls for two optimizations (easy access to the PID + microsecond timestamp precision), both of which are indeed optional (the driver will detect their absence and has non-native fallbacks available for both cases).  You'll get debug-level slf4j stack traces but that should be the worst consequence.

The jackson-databind dependency is used for parsing an XML config in the Astra connectivity code.  This can also be excluded for the non-Astra case (which presumably is the default).

A hearty 👍 to the use of the shaded JAR as well.

I have a simple test app that builds and runs without issue using the following Gradle dependency:

```
    compile('com.datastax.cassandra:cassandra-driver-core:3.11.2:shaded') {
        exclude group:'io.netty', module:'netty-handler'
        exclude group:'com.github.jnr', module:'jnr-ffi'
        exclude group:'com.github.jnr', module:'jnr-posix'
        exclude group:'com.fasterxml.jackson.core', module:'jackson-databind'
    }
```

Reply:
@absurdfarce thanks for jumping on this PR. And thanks for the confirmation and recommendations. I'll do what you both recommend.

Reply:
@zentol a question related to netty: I went through [FLINK-8295](https://issues.apache.org/jira/browse/FLINK-8295) and in this ticket we relocate netty to `com.datastax.shaded.netty` so that cassandra driver could find it. But it would have found it also with default `io.netty` so why can't we just keep netty transitive dep as it is ? Is it conflicting with netty elsewhere in flink ? (I checked the version used in cassandra is the same as in flink parent)

Reply:
done